src/OpenLoco/Map/MapGenerator.cpp

```diff
@@ -427,14 +430,199 @@ namespace OpenLoco::Map::MapGenerator
         }
     }
 
-    // 0x0046A021
-    static void generateTerrain(HeightMap& heightMap)
+    static std::optional<uint8_t> getEverywhereSurfaceStyle()
+    {
+        for (uint8_t landObjectIdx = 0; landObjectIdx < ObjectManager::getMaxObjects(ObjectType::land); ++landObjectIdx)
+        {
+            auto* landObj = ObjectManager::get<LandObject>(landObjectIdx);
+            if (landObj == nullptr)
+            {
+                continue;
+            }
+            if (S5::getOptions().landDistributionPatterns[landObjectIdx] == LandDistribuitionPattern::everywhere)
+            {
+                return landObjectIdx;
+            }
+        }
+        if (*_primaryLandObjectIndex != 0xFF)
+        {
+            return *_primaryLandObjectIndex;
+        }
+        return std::nullopt;
+    }
+
+    // 0x00469FC8
+    std::optional<uint8_t> getRandomTerrainVariation(const SurfaceElement& surface)
+    {
+        if (surface.water())
+        {
+            return std::nullopt;
+        }
+        if (surface.hasHighTypeFlag())
+        {
+            return std::nullopt;
+        }
+
+        auto* landObj = ObjectManager::get<LandObject>(surface.terrain());
+        if (landObj == nullptr)
+        {
+            return std::nullopt;
+        }
+
+        if (landObj->var_1B == 0 || surface.slope())
+        {
+            return 0;
+        }
+
+        // TODO: split into two randNext calls
```

Comment:
still todo?

Reply:
Yeah will cause a desync otherwise. So will leave it until all the caller's of the function are implemented

Reply:
no worries, just checking if it needed to be implemented in this PR, or its ok to do in future as a refactor